www/js/survey/multilabel/MultiLabelButtonGroup.tsx

```diff
@@ -2,75 +2,62 @@
   In the default configuration, these are the "Mode" and "Purpose" buttons.
   Next to the buttons is a small checkmark icon, which marks inferrel labels as confirmed */
 
-import React, { useContext, useEffect, useState } from "react";
-import { angularize, createScopeWithVars, getAngularService } from "../../angular-react-helper";
-import { object, number } from "prop-types";
-import { View } from "react-native";
-import { IconButton, Text, useTheme } from "react-native-paper";
+import React, { useContext, useEffect, useState, useMemo } from "react";
+import { getAngularService } from "../../angular-react-helper";
+import { View, Modal, ScrollView, Pressable, useWindowDimensions } from "react-native";
+import { IconButton, Text, Dialog, useTheme, RadioButton, Button, TextInput } from "react-native-paper";
 import DiaryButton from "../../diary/DiaryButton";
 import { useTranslation } from "react-i18next";
 import { LabelTabContext } from "../../diary/LabelTab";
+import { displayErrorMsg, logDebug } from "../../plugin/logger";
+import { getLabelInputDetails, getLabelInputs, getLabelOptions } from "./confirmHelper";
 
 const MultilabelButtonGroup = ({ trip }) => {
   const { colors } = useTheme();
   const { t } = useTranslation();
   const { repopulateTimelineEntry } = useContext(LabelTabContext);
+  const { height: windowHeight } = useWindowDimensions();
 
   const [ inputParams, setInputParams ] = useState({});
-  let closePopover;
-
-  const ConfirmHelper = getAngularService("ConfirmHelper");
-  const $ionicPopup = getAngularService("$ionicPopup");
-  const $ionicPopover = getAngularService("$ionicPopover");
+  // modal visible for which input type? (mode or purpose or replaced_mode, null if not visible)
+  const [ modalVisibleFor, setModalVisibleFor ] = useState<'MODE'|'PURPOSE'|'REPLACED_MODE'|null>(null);
+  const [otherLabel, setOtherLabel] = useState<string|null>(null);
+  const chosenLabel = useMemo<string>(() => {
+    if (otherLabel != null) return 'other';
+    return trip.userInput[modalVisibleFor]?.value
+  }, [modalVisibleFor, otherLabel]);
 
   useEffect(() => {
     console.log("During initialization, trip is ", trip);
-    ConfirmHelper.inputParamsPromise.then((ip) => setInputParams(ip));
+    getLabelOptions().then((ip) => setInputParams(ip));
   }, []);
 
   // to mark 'inferred' labels as 'confirmed'; turn yellow labels blue
   function verifyTrip() {
-    for (const inputType of ConfirmHelper.INPUTS) {
+    for (const inputType of getLabelInputs()) {
       const inferred = trip.finalInference[inputType];
       // TODO: figure out what to do with "other". For now, do not verify.
       if (inferred?.value && !trip.userInput[inputType] && inferred.value != "other")
         store(inputType, inferred.value, false);
```

Comment:
> If the user selects other, we generate a popup to enter the new mode. We might be able to simplify this in the next release?

Already simplified. If the user selects "Other", a text box immediately appears below "Other" instead of a separate popup. This is stored and can be selected from the list on subsequent labelings

Reply:
Removed extraneous check in https://github.com/e-mission/e-mission-phone/pull/1023